balancer-js/src/pool-stable/calculate-invariant.ts

```diff
@@ -0,0 +1,84 @@
+import { SolidityMaths } from '@/lib/utils/solidityMaths';
+import { AMP_PRECISION } from '@/lib/utils';
+import { BalancerError, BalancerErrorCode } from '@/balancerErrors';
+
+/**********************************************************************************************
+ // invariant                                                                                 //
+ // D = invariant                                                  D^(n+1)                    //
+ // A = amplification coefficient      A  n^n S + D = A D n^n + -----------                   //
+ // S = sum of balances                                             n^n P                     //
+ // P = product of balances                                                                   //
+ // n = number of tokens                                                                      //
+ **********************************************************************************************/
+
+export const calculateStableInvariant = (
```

Comment:
Same here about the [SOR](https://github.com/balancer-labs/balancer-sor/blob/master/src/pools/stablePool/stableMathBigInt.ts#L15)